src/main/java/organice/logic/parser/ProcessingMarkDoneCommandParser.java

```diff
@@ -0,0 +1,44 @@
+package organice.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+
+import static organice.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static organice.logic.parser.CliSyntax.PREFIX_NRIC;
+
+import organice.logic.commands.ProcessingCommand;
+import organice.logic.commands.ProcessingMarkDoneCommand;
+import organice.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new FindCommand object
+ */
+public class ProcessingMarkDoneCommandParser implements Parser<ProcessingMarkDoneCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ProcessingCommand
+     * and returns a ProcessingCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ProcessingMarkDoneCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        String trimmedArgs = args.trim();
+        String prefixNricString = PREFIX_NRIC.toString();
+
+        String firstNric;
+        String secondNric;
+        String taskNumber;
+
+        String[] nameKeywords = trimmedArgs.split("\\s+");
+
+        if (nameKeywords[0].startsWith(prefixNricString) && nameKeywords[1].startsWith(prefixNricString)) {
+            firstNric = nameKeywords[0].replaceFirst(prefixNricString, "");
+            secondNric = nameKeywords[1].replaceFirst(prefixNricString, "");
+            taskNumber = nameKeywords[2];
```

Comment:
Everytime u split like this, u should try catch the `ArrayIndexOutOfBounds` i think right now if u type `processingMarkDone ic/usahfdushdfu` it will crash the app

Reply:
amended